thavalon-server/src/game/state/on_mission.rs

```diff
@@ -118,6 +119,36 @@ impl GameState<OnMission> {
     fn includes_player(&self, player: &str) -> bool {
         self.proposal().players.contains(player)
     }
+
+    // Append messages for lovers to an exisiting list of effects. This will not modify effects if there are no lovers in game,
+    // so this is safe to call in any game.
+    fn add_lover_effects(&self, effects: &mut Vec<Effect>) {
+        let tristan_display_name = self.game.display_name_from_role(Role::Tristan);
+        let tristan_on_mission = tristan_display_name.map_or(None, |x| Some(self.proposals.get(self.phase.proposal_index).unwrap().players.contains(x)));
+        let iseult_display_name = self.game.display_name_from_role(Role::Iseult);
+        let iseult_on_mission = iseult_display_name.map_or(None, |x| Some(self.proposals.get(self.phase.proposal_index).unwrap().players.contains(x)));
+        // First, add messages going to Tristan. If there is no Tristan, do nothing.
+        if tristan_display_name.is_some() {
+            effects.push(self.build_lover_effect(tristan_display_name.unwrap(), tristan_on_mission.unwrap(), iseult_display_name, iseult_on_mission));
+        }
```

Comment:
Instead of using `.is_some()` and `.unwrap()`, we can use `if let`:

```
if let Some(tristan_display_name) = tristan_display_name {
  effects.push(...)
}
```